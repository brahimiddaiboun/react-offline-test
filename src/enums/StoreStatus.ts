export enum StoreStatus {
  OFFLINE,
  ONLINE,
  ERROR,
  STALE,
  LOADING,
}
