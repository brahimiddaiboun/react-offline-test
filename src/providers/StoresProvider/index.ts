export { storesContext, StoresProvider } from './StoresProvider'
