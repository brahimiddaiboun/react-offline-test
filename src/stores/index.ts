import { configure } from "mobx";

import memoize from "memoize-state";

import { EnergyStore } from "./EnergyStore";

configure({
  isolateGlobalState: true,
});

interface StoreConfig {}

export type RootStore = {
  energyStore: EnergyStore;
};

type StoreCreator = (config: StoreConfig) => RootStore;

export const createStore: StoreCreator = memoize(() => ({
  energyStore: new EnergyStore(),
}));

export { EnergyStore };
