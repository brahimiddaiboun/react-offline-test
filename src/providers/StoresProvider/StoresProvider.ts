import React from 'react'
import { createStore } from '../../stores'

export const storesContext = React.createContext(createStore({}))
export const StoresProvider = storesContext.Provider
