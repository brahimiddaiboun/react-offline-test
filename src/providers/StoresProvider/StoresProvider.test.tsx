import React, { useContext } from 'react'

import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { StoresProvider, storesContext } from './StoresProvider'
import { EnergyStore } from '../../stores'
import { StoreStatus } from '../../enums'

describe('StoresProvider', () => {
    it('storesContext should be a React context', () => {
        expect(storesContext.Provider).toBeDefined()
    })

    it('should be available in context', () => {
        const TestComponent: React.FC = () => {
            const { energyStore } = useContext(storesContext)
            energyStore.fetchEnergyGeneration()
            if (energyStore.status === StoreStatus.ERROR) {
                return <title data-testid="error">error</title>
            }
            return null
        }

        const reactElement: React.ReactElement = (
            <StoresProvider value={{ energyStore: new EnergyStore() }}>
                <TestComponent />
            </StoresProvider>
        )

        const testReturnRender = render(reactElement)
        expect(testReturnRender.getAllByTestId('error')).toBeDefined()

        const tree = renderer.create(reactElement)
        expect(tree.toJSON()).toMatchSnapshot()
    })
})
