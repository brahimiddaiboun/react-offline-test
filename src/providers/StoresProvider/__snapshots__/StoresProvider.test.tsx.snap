// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoresProvider should be available in context 1`] = `
<title
  data-testid="error"
>
  error
</title>
`;
